process_bug.cgi

```diff
@@ -73,23 +74,36 @@ if (Bugzilla->params->{disable_bug_updates}) {
 
 # Create a list of objects for all bugs being modified in this request.
 my @bug_objects;
+my @bug_ids;
+
+# $async_bulk_edit is whether or not we use the bulk editing code,
+# $background is if the background=1 parameter was passed.
+my $async_bulk_edit = $cgi->param('async_bulk_edit') // 0;
+$cgi->delete('async_bulk_edit') if $async_bulk_edit;
+
 if (defined $cgi->param('id')) {
   my $bug = Bugzilla::Bug->check(scalar $cgi->param('id'));
   $cgi->param('id', $bug->id);
   push(@bug_objects, $bug);
 }
 else {
-  foreach my $i ($cgi->param()) {
-    if ($i =~ /^id_([1-9][0-9]*)/) {
-      my $id = $1;
-      push(@bug_objects, Bugzilla::Bug->check($id));
-    }
+  @bug_ids = map { /^id_([1-9][0-9]*)/ ? $1 : () } $cgi->param;
+
+  # Make sure there are bugs to process.
+  ThrowUserError("no_bugs_chosen", {action => 'modify'}) unless @bug_ids;
+  $async_bulk_edit = 1 if @bug_ids > 10;
```

Comment:
Just noticed this. Seems that even if the user does not check the bulk edit checkbox on buglist.cgi, it will still enable it if @bug_ids > 10 which negates the need for a checkbox altogether.

Maybe this should be instead:

$async_bulk_edit = 0 if @bug_ids < 10; 

Or am I missing something?

Reply:
right now, doing > 10 edits is likely to fail, so it's meant to make them more likely to succeed. Eventually bz_async_bulk_edit should be the same as editbugs. Does that make sense?

Reply:
Sounds fine. But is still sort of misleading from a user standpoint. Maybe we should add some text or tooltip that states it is automatic if more than 10 bugs?